### PR TITLE
Fixed Duplicate Source Name and Email Check

### PR DESCRIPTION
**Pull request by Ram6911**

### CODE REVIEW

File: microsetta_interface/templates/new_participant.jinja2 [hunk 1]
```diff
@@ -249,15 +294,38 @@
...
     </form>
 </div>
-{% endblock %}
+
+<!-- Modal -->
+<div class="modal fade" id="source_modal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLabel">Message</h5>
```
Comment (Ram6911) [inline microsetta_interface/templates/new_participant.jinja2:314]: Change name Message to Warning.

File: microsetta_interface/implementation.py [hunk 2]
```diff
@@ -784,22 +783,36 @@ def post_account_details(*, account_id=None, body=None):
     return _refresh_state_and_route_to_sink(account_id)
 
 
+@prerequisite([ACCT_PREREQS_MET])
+def source_duplicate_email_check(*, account_id=None, body=None):
```
Comment (cassidysymons) [inline microsetta_interface/implementation.py:787]: Can you please rename this function to be more clear about the fact that it's not just an email duplicate check?

File: microsetta_interface/implementation.py [hunk 3]
```diff
@@ -784,22 +783,36 @@ def post_account_details(*, account_id=None, body=None):
...
+    has_error, email_check_output, _ = ApiRequest.post(
```
Comment (cassidysymons) [inline microsetta_interface/implementation.py:788]: Need to add a condition to handle the possibility that has_error has a value here.

File: microsetta_interface/routes.yaml [hunk 4]
```diff
@@ -350,6 +350,27 @@ paths:
               schema:
                 type: string
 
+  '/accounts/{account_id}/source_duplicate_email_check':
```
Comment (cassidysymons) [inline microsetta_interface/routes.yaml:353]: Please rename this path to be more clear about its use.

File: microsetta_interface/templates/new_participant.jinja2 [hunk 5]
```diff
@@ -108,6 +103,57 @@
...
         disableInputOnSubmit("#18-form", "#18-submit_button", "{{ _('Saving...') }}");
+        $("#home_page").click(function(){
+            window.open('{{home_url}}', '_self')
+        })
+        $("#create_source").click(function(){
+            //window.open('{{post_url}}', '_self')
+            //$("#source_modal").modal('hide');
+            $("#18-form")[0].submit();
+        })
+        $("#18-form").submit(function(e){
```
Comment (cassidysymons) [inline microsetta_interface/templates/new_participant.jinja2:114]: This needs to work for all four types of consent: 0-6, 7-12, 13-17, and 18+. Currently it only works for 18+.

File: microsetta_interface/templates/new_participant.jinja2 [hunk 6]
```diff
@@ -249,15 +294,38 @@
...
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary" id="home_page">Return to Home Page</button>
+        <button type="button" class="btn btn-primary" id="create_source">Proceed with Creating New Source</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+
+{% endblock %}
```
Comment (cassidysymons) [inline microsetta_interface/templates/new_participant.jinja2:331]: Please add a newline at the end of the file.

**Review by cassidysymons — CHANGES_REQUESTED** [review]

File: microsetta_interface/implementation.py [hunk 7]
```diff
@@ -784,22 +783,39 @@ def post_account_details(*, account_id=None, body=None):
     return _refresh_state_and_route_to_sink(account_id)
 
 
+@prerequisite([ACCT_PREREQS_MET])
+def check_duplicate_source_name_email(*, account_id=None, body=None):
+    has_error, email_check_output, _ = ApiRequest.post(
+        "/accounts/{0}/check_duplicate_source".format(account_id), json=body)
+    if has_error:
+        return consent_output
```
Comment (cassidysymons) [inline microsetta_interface/implementation.py:791]: consent_output isn't defined anywhere in this function - did you mean email_check_output?

File: microsetta_interface/templates/new_participant.jinja2 [hunk 8]
```diff
@@ -108,6 +120,61 @@
...
+                valid = validate1317("13-17-form")
+            } else if ($form_id == "18-form") {
+                valid = validate18("18-form")
+            }
+
+            var duplicate_source_check_msg = `It looks like you're creating a new source that may
+                  be similar or the same as an existing source.
+                  If this is the same person as an existing source,
+                  please consider providing new samples or survey
+                  responses under that source`;
```
Comment (cassidysymons) [inline microsetta_interface/templates/new_participant.jinja2:148]: Can you please wrap this string in gettext() shorthand ( {{ _('It looks like...') }} ) so PyBabel can pick it up for translation?

**Review by cassidysymons — CHANGES_REQUESTED** [review]

File: microsetta_interface/implementation.py [hunk 9]
```diff
@@ -788,7 +788,7 @@ def check_duplicate_source_name_email(*, account_id=None, body=None):
     has_error, email_check_output, _ = ApiRequest.post(
         "/accounts/{0}/check_duplicate_source".format(account_id), json=body)
     if has_error:
-        return consent_output
+        email_check_output["error"] = has_error
```
Comment (cassidysymons) [inline microsetta_interface/implementation.py:791]: This line should be "return email_check_output"

**Review by cassidysymons — CHANGES_REQUESTED** [review]

**Review by cassidysymons — CHANGES_REQUESTED** [review]
Hey Ram, I'm testing this on our staging platform, and when you select "7 - 12 years" or "13 - 17 years" on the consent screen, it shows both of those consent forms, rather than just the selected one. Can you take a look at that and submit a fix? Thanks!

Edit: This was a transient issue in my browser, can be ignored.

**Review by cassidysymons — APPROVED** [review]

**Pull request merged**